### DISCUSSION TITLE
Decimals on ERC20

### DISCUSSION THREAD

**Discussion by martriay** [OpenZeppelin/cairo-contracts#81] [Design]
Considering that:
* In the [solidity implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L48-L49) of the contract the value is hardcoded at `18` and the way to change it is through overloading, which is not available in Cairo.
* According to [EIP20](https://eips.ethereum.org/EIPS/eip-20) `decimals`' type should be `uint8`, which is way smaller than Cairo's `felt`. This could cause problems when bridging tokens to EVM compatible chains.

Should we then add a new parameter to the constructor and sanitize it's under `2^8`?

**Comment by frangio** [comment]
Most tokens should use 18, so I think we should just set it to 18.

**Reply by martriay** [reply]
Yeah but some don't (e.g. [USDC](https://etherscan.io/token/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48?a=0x970b6f7c17f52f6b7be61fc334840b78b73f0aa3) uses `6`). We should aim for maximal compatibility.
